src/inverted_index.c

```diff
@@ -226,32 +226,35 @@ ENCODER(encodeRawDocIdsOnly) {
 
 #define NUM_TINYENC_MASK 0x07  // This flag is set if the number is 'tiny'
 
+#define NUM_ENCODING_COMMON_TYPE_TINY           0
+#define NUM_ENCODING_COMMON_TYPE_FLOAT          1
+#define NUM_ENCODING_COMMON_TYPE_POSITIVE_INT   2
+#define NUM_ENCODING_COMMON_TYPE_NEG_INT        3
+
 typedef struct {
-  uint8_t deltaEncoding : 2;
-  uint8_t zero : 2;
-  uint8_t valueByteCount : 3;
-  uint8_t sign : 1;
+  uint8_t deltaEncoding : 3;
+  uint8_t type : 2;
+  uint8_t valueByteCount : 3; //1 to 8 (encoded as 0-7, since value 0 is represented as tiny)
```

Comment:
Can you explain this change? Also lets add a comment what each field means and maybe some general comment about how we encode?
If you can add to the top comment some general explanation about the encoding changes it could also be helpfull.